augur/generate.py

```diff
@@ -133,15 +150,43 @@ def two_point_template(config):
             print("Missing Nz_type in %s. Quitting." % src)
             raise RuntimeError
         if tcfg["Nz_type"] == "Gaussian":
-            mu, sig = tcfg["Nz_center"], tcfg["Nz_sigma"]
-            zar = np.linspace(max(0, mu - 5 * sig), mu + 5 * sig, 500)
-            Nz = np.exp(-((zar - mu) ** 2) / (2 * sig ** 2))
-            S.add_tracer("NZ", src, zar, Nz)
+            mu, wi = tcfg["Nz_center"], tcfg["Nz_width"]
+            alpha, z0 = tcfg["Nz_alpha"], tcfg["Nz_z0"]
+            sig_z = tcfg["Nz_sigmaz"]
+            zall = np.linspace(0, 4, 1500)
+            mask = (zall > mu-wi/2) & (zall < mu+wi/2)
+            dndz_bin = np.zeros_like(zall)
+            dndz_bin[mask] = srd_dndz(zall[mask], z0, alpha)
+            # Convolve the SRD N(z) with a Gaussian with the required smearing
+            Nz = np.array([np.exp(-0.5*(zk-zall)**2/((sig_z*(1+zk))**2))*dndz_bin for zk in zall])
```

Comment:
Could you do here something like `scipy.ndimage.gaussian_filter` or something?

Reply:
I checked and there are a couple of caveats. The first one is that it doesn't allow you to use a kernel with varying sigma (in this case I'm doing `sigma = sig_z*(1+z)`). The second problem is that if you just go and set `sigma = 0.05*(1+zcent)` for the Gaussian filter it won't work. The filter sigma has to be specified in units of the step in the `zall` array that I'm using there (so itd' be `0.05*(1+zcent)/(zall[1]-zall[0])`). It is true that the convolution is ~30 times faster but I think this is a one time operation (and also, it only takes 30 ms in the slow version).

Reply:
Fine, then leave it as it is.